orchestration/templates/diff-table.yaml

```diff
@@ -61,10 +142,88 @@ spec:
           - name: GCS_PREFIX
             value: '{{ "gs://{{inputs.parameters.gcs-bucket}}/{{inputs.parameters.input-prefix}}" }}'
           - name: OUTPUT_SUFFIX
-            value: '{{ template "argo.timestamp" }}'
+            value: '{{ "{{inputs.parameters.output-table-suffix}}" }}'
         command: [bash]
         source: |
           {{- /* More verbose than I expected, but direct injection is too hard to format. */}}
           {{- range .Files.Lines "scripts/diff-bq-table.sh" }}
           {{ . }}
           {{- end }}
+
+    - name: query-ids-to-delete
+      inputs:
+        parameters:
+          - name: table-name
+          - name: join-table-name
+          - name: bq-project
+          - name: bq-dataset
+          - name: output-table-suffix
+      script:
+        image: us.gcr.io/broad-dsp-gcr-public/clinvar-schema:{{ default "latest" .Chart.AppVersion }}
+        env:
+          - name: PROJECT
+            value: '{{ "{{inputs.parameters.bq-project}}" }}'
+          - name: DATASET
+            value: '{{ "{{inputs.parameters.bq-dataset}}" }}'
+          - name: INPUT_TABLE
+            value: '{{ "{{inputs.parameters.join-table-name}}" }}'
+          - name: OUTPUT_PREFIX
+            value: '{{ "{{inputs.parameters.table-name}}" }}'
+          - name: OUTPUT_SUFFIX
+            value: '{{ "{{inputs.parameters.output-table-suffix}}" }}'
+        command: [bash]
+        source: |
+          {{- range .Files.Lines "scripts/query-non-null-row-ids.sh" }}
```

Comment:
I know you have this pattern already but it caught my eye more clearly this time; I think this is readable enough, but I wonder if creating a [partial template](https://helm.sh/docs/chart_template_guide/named_templates/) would help with the reusability of this piece? Since it is such a simple function it might end up looking less readable and not be worth the hassle, but might be worth considering.

Reply:
I tried it out, but I hit a roadblock. The named templates apparently don't have access to the global `values` scope, so if you want to call `.Files.Lines` within one then you have to pass that root context to the `include` call. That's a problem because for the template to be generic, you also need to pass in the script name as context. To make that work you either need to `tuple` or `dict` together the root context and the script name, which turns into a bunch of boilerplate on its own.